src/main/java/com/mongodb/kafka/connect/util/ConfigHelper.java

```diff
@@ -126,4 +132,43 @@ public static String getOverrideOrDefault(
     }
     return stringConfig;
   }
+
+  @SuppressWarnings("deprecated")
+  public static void logDeprecationWarnings(
+      Map<String, String> deprecatedPropertiesWithReplacements, Map<String, Object> originals) {
```

Comment:
`Map` seems unnecessary here, can we just have `String deprecatedProperty, String replacement` as arguments to the function, and call it multiple times?
And perhaps we could combine this with readPropertyWithDeprecatedFallback if we stored the result and only called that function once per config.

Reply:
> And perhaps we could combine this with readPropertyWithDeprecatedFallback if we stored the result and only called that function once per config.

I considered something like this but didn't like the implications w/r/t modifying the output when the config class logs its values. Seems like there's value in preserving (and logging) a record of what exactly the user specified, including which variants (or both, or neither) of the `errors.tolerance` and related properties were supplied.

Fine with changing this from a map to a more focused method 👍